src/rt/arc.rs

```diff
@@ -121,21 +121,22 @@ impl State {
         assert_eq!(0, self.ref_cnt, "Arc leaked");
     }
 
-    pub(super) fn last_dependent_accesses<'a>(
-        &'a self,
+    pub(super) fn for_each_last_dependent_access(
```

Comment:
I **believe** that returning an iterator is no longer necessary. This can **probably** be changed to:

```rust
fn last_dependent_access(&self, action: Action) -> Option<&Access> { ... }
```

Reply:
Yes, every object state except Atomic doesn't need an iterator. Atomic still needs to iterate over `last_load` and `last_store` though. But I like the idea of simply returning an Option for every other object - the code is simpler and it is even a bit faster that way :)

Reply:
ah, i'm actually pretty sure that can be removed... Instead of separate load / store access values, there can be a single one for the entire atomic (the reason there are multiple is a failed attempt at an optimization).

Reply:
Do you want to open a separate PR to do ^^ then we can update this one.

Reply:
https://github.com/tokio-rs/loom/pull/98